### PR TITLE
Vungle mediation adapter 6.11.0.0 release

### PR DESCRIPTION
**Pull request by ManojBudumuru-vungle**
- Verified compatibility with Vungle SDK 6.11.0
- Native ads support.
- Coppa Support.

### CODE REVIEW

**Comment by stanley-vungle** [comment]
@ericleich Can we have help to resolve this CLA issue?
@ManojBudumuru-vungle has followed the steps and accepted the agreement, but it keeps failing. Is there something that needs to be done on your end?

**Comment by joshh-devrel** [comment]
@ManojBudumuru-vungle Can you rebase this pull request?

**Comment by ManojBudumuru-vungle** [comment]
> @ManojBudumuru-vungle Can you rebase this pull request?

@joshh-devrel Done.

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleDelegate.h [hunk 1]
```diff
@@ -39,6 +39,7 @@ typedef NS_ENUM(NSUInteger, BannerRouterDelegateState) {
 - (void)adAvailable;
 - (void)adNotAvailable:(nonnull NSError *)error;
 - (void)willShowAd;
+- (void)didShowAd;
```
[the remark this answers is not in the document]
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/GADMAdapterVungleDelegate.h:42]: No. It has been there for a while. This was added to separate the rewarded ad callback `didStartVideo` from the `willPresentFullScreenView`. This was requested in one of the video meetings some time ago.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GADMediationVungleNativeAd.h"
+#include <stdatomic.h>
+#import "GADMAdapterVungleUtils.h"
+#import "GADMAdapterVungleRouter.h"
+#import <VungleSDK/VungleNativeAd.h>
+
+@interface GADMediationVungleNativeAd () <GADMediationNativeAd, VungleNativeAdDelegate, GADMAdapterVungleDelegate>
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:21]: Isn't that for native we only support waterfall mediation (not bidding)?

Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:21]: Yes. Both flows will utilize the same class. There is no difference between the two.
The native ad implementation on Vungle's end utilizes an instance-based approach so it doesn't need the router to track its reference to relay callbacks to the ad object.

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m [hunk 3]
```diff
@@ -419,6 +421,13 @@ - (NSString *)getSuperToken {
   return [VungleSDK.sharedSDK currentSuperToken];
 }
 
+- (void)setCOPPAStatus:(NSNumber *)coppa {
+    if (coppa) {
+        [[VungleSDK sharedSDK] updateCOPPAStatus:[coppa boolValue]];
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m:426]: Seems like the publishers don't need to use another Vungle API to set the COPPA status, but the adapter can just get what's set through the Google Mobile Ads SDK and pass it to Vungle SDK. Is that correct?

Reference:
- https://developers.google.com/admob/ios/targeting#child-directed_setting
- https://developers.google.com/admob/ios/api/reference/Classes/GADMediationAdConfiguration?hl=en#childdirectedtreatment

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.h [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+#import <Foundation/Foundation.h>
+#import <GoogleMobileAds/GoogleMobileAds.h>
+
+@interface GADMediationVungleNativeAd : NSObject
+
+/// Asks the receiver to render the ad configuration.
+- (nonnull instancetype)initNativeAdForAdConfiguration:(nonnull GADMediationNativeAdConfiguration *)adConfiguration
+                                     completionHandler:(nonnull GADMediationNativeLoadCompletionHandler)completionHandler;
+- (nonnull instancetype)init NS_UNAVAILABLE;
+- (void)requestAd;
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.h:24]: Nit, prefer: `requestNativeAd`.

Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.h:24]: Nit, prefer adding comments.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+        delegate = originalCompletionHandler(ad, error);
+      }
+      originalCompletionHandler = nil;
+      return delegate;
+    };
+  }
+  return self;
+}
+
+- (void)requestAd {
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:71]: Nit, `requestNativeAd`.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+- (nullable UIView *)adChoicesView {
+  return nil;
+}
+
+- (nullable UIView *)mediaView {
+  return _mediaView;
+}
+
+- (BOOL)hasVideoContent {
+  return YES;
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:150]: Is it guarantee that the native ad has video content everytime?

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+- (nullable NSString *)price {
+  return nil;
+}
+
+- (nullable NSString *)advertiser {
+  return _nativeAd.sponsoredText;
+}
+
+- (nullable NSDictionary<NSString *, id> *)extraAssets {
+  return nil;
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:138]: Looking at:
https://support.vungle.com/hc/en-us/articles/4416477036955#display-a-native-ad-0-5

There is "Sponsored By" text, does the adapter want to handle that as well?

Also see [`extraAssets`](https://developers.google.com/admob/ios/api/reference/Protocols/GADMediatedUnifiedNativeAd?hl=en#extraassets).

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+}
+
+#pragma mark - GADMediatedUnifiedNativeAd
+
+- (nullable NSString *)headline {
+    return _nativeAd.title;
+}
+
+- (nullable NSArray<GADNativeAdImage *> *)images {
+  return nil;
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:106]: If this is `nil`, I would assume Vungle only returns video content for native ad?

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+- (nullable NSString *)advertiser {
+  return _nativeAd.sponsoredText;
+}
+
+- (nullable NSDictionary<NSString *, id> *)extraAssets {
+  return nil;
+}
+
+- (nullable UIView *)adChoicesView {
+  return nil;
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:142]: Just to confirm, there isn't ad choice?

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+    clickableAssetViews:(nonnull NSDictionary<GADNativeAssetIdentifier, UIView *> *)clickableAssetViews
+ nonclickableAssetViews:(nonnull NSDictionary<GADNativeAssetIdentifier, UIView *> *)nonclickableAssetViews
+         viewController:(nonnull UIViewController *)viewController {
+  NSArray<UIView *> *assets = clickableAssetViews.allValues;
+  UIImageView *iconView = nil;
+  if ([clickableAssetViews[GADNativeIconAsset] isKindOfClass:[UIImageView class]]) {
+    iconView = (UIImageView *)clickableAssetViews[GADNativeIconAsset];
+  }
+    
+  [_nativeAd registerViewForInteraction:view mediaView:_mediaView iconImageView:iconView viewController:viewController clickableViews:assets];
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:163]: Would you mind clarifying how the Vungle SDK tracks clicks and impressions?

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m [hunk 11]
```diff
@@ -0,0 +1,248 @@
...
+    NSString *appID = [GADMAdapterVungleUtils findAppID:_adConfiguration.credentials.settings];
+    [[GADMAdapterVungleRouter sharedInstance] initWithAppId:appID delegate:self];
+    return;
+  }
+    
+  [self loadAd];
+}
+
+- (void)loadAd {
+  _nativeAd = [[VungleNativeAd alloc] initWithPlacementID:self.desiredPlacement];
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:90]: I would assume if we support native bidding, bid response will be passed here correct?

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m (hunk 3 above)
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m:426]: Yes. That's correct. The publisher never uses the Router class. The Router class is a wrapper to communicate with the Vungle SDK. I'm following the implementation guidelines for this adapter with this function.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m (hunk 6 above)
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:150]: No.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m [hunk 12]
```diff
@@ -0,0 +1,248 @@
...
+- (void)didCloseAd {
+    // Do nothing. Native ads utilize a different set of callbacks.
+}
+
+- (void)willLeaveApplication {
+    // Do nothing. Native ads utilize a different set of callbacks.
+}
+
+- (nullable NSString *)bidResponse {
+    return [_adConfiguration bidResponse];
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:245]: Seems like this is no-op and will return `null` everytime?

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleRewardedAd.m [hunk 13]
```diff
@@ -143,18 +143,18 @@ - (void)adAvailable {
 
 - (void)didCloseAd {
   [_delegate didDismissFullScreenView];
-
-  [GADMAdapterVungleBiddingRouter.sharedInstance removeDelegate:self];
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleRewardedAd.m:147]: Why this is removed?

We still have `addDelegate` in the bidding router:
https://github.com/googleads/googleads-mobile-ios-mediation/blob/master/adapters/Vungle/VungleAdapter/Bidding/GADMAdapterVungleBiddingRouter.m#L73-L81

File: adapters/Vungle/Public/Headers/VungleAdNetworkExtras.h [hunk 14]
```diff
@@ -43,4 +43,6 @@
 
 @property (nonatomic, readonly, assign) BOOL muteIsSet;
 
+@property (nonatomic, assign) NSInteger nativeAdOptionPosition;
```
Comment (joshh-devrel) [inline adapters/Vungle/Public/Headers/VungleAdNetworkExtras.h:46]: Should there be a `NS_ENUM` for what the publisher can set?

See step 4 in https://support.vungle.com/hc/en-us/articles/4416477036955#configure-the-native-ad-for-display-using-storyboard-0-6.

And I would assume the publisher can set this via the code below (referring to [this dev doc](https://developers.google.com/admob/ios/mediation/vungle#network-specific_parameters)):
```
#import <VungleAdapter/VungleAdapter.h>
//...

GADRequest *request = [GADRequest request];
VungleAdNetworkExtras *extras = [[VungleAdNetworkExtras alloc] init];
extras.nativeAdOptionPosition = 1; // Specify the position of the privacy icon to be in the top left.
//...
[request registerAdNetworkExtras:extras];
```

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m [hunk 15]
```diff
@@ -0,0 +1,248 @@
...
+  [self loadAd];
+}
+
+- (void)loadAd {
+  _nativeAd = [[VungleNativeAd alloc] initWithPlacementID:self.desiredPlacement];
+  _nativeAd.delegate = self;
+  VungleAdNetworkExtras *networkExtras = [_adConfiguration extras];
+  if (networkExtras.nativeAdOptionPosition != 0) {
+    _nativeAd.adOptionsPosition = networkExtras.nativeAdOptionPosition;
+  }
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:95]: I don't think the `!= 0` check is sufficient here, publishers might set an unexpected value (like 100).

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m [hunk 16]
```diff
@@ -0,0 +1,248 @@
...
+- (void)initialized:(BOOL)isSuccess error:(nullable NSError *)error {
+  if (isSuccess) {
+    [self loadAd];
+  } else {
+    _adLoadCompletionHandler(nil, error);
+  }
+}
+
+- (void)adAvailable {
+    // Do nothing. Native ads utilize a different set of callbacks.
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:205]: Nit, prefer:
```
// No-op, Vungle native ads utilize callbacks from VungleNativeAdDelegate.
```

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m (hunk 7 above)
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:138]: Isn't that the `advertiser` property above?

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m (hunk 9 above)
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:142]: No public API for it. The "ad choice" is the icon for redirecting to the privacy, is that correct? That's what we understood what that term meant and our product team decided to handle the display of the icon internally.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m (hunk 6 above)
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:150]: If I set this to `NO`, then the `mediaView` was unable to render.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m (hunk 10 above)
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:163]: Impression callback is fired when the ad is rendered. The click callback is fired when the user taps on the ad. The callbacks are handled in this ad class in the `VungleNativeAdDelegate` section below. The class is registered as the native ad delegate at L91.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m (hunk 11 above)
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:90]: Yes. It should be set in the object initialization.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleRewardedAd.m (hunk 13 above)
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleRewardedAd.m:147]: Because this is a duplicate callback. It is handled after the `didCloseAd` call in the router.

https://github.com/googleads/googleads-mobile-ios-mediation/blob/master/adapters/Vungle/VungleAdapter/Bidding/GADMAdapterVungleBiddingRouter.m#L210

File: adapters/Vungle/Public/Headers/VungleAdNetworkExtras.h (hunk 14 above)
Comment (stanley-vungle) [reply adapters/Vungle/Public/Headers/VungleAdNetworkExtras.h:46]: Yes, the example is correct. The publisher would set this through the network extras.
I didn't add an enum to the adapter code and followed the implementation of the `orientation` property, which uses the integer and translates to our enum value.

https://github.com/googleads/googleads-mobile-ios-mediation/blob/master/adapters/Vungle/Public/Headers/VungleAdNetworkExtras.h#L40

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m (hunk 15 above)
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:95]: I will address this in another commit. It is in internal review currently.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m (hunk 16 above)
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:205]: I will address this in another commit. It is in internal review currently.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m (hunk 5 above)
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:71]: I will address this in another commit. It is in internal review currently.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.h (hunk 4 above)
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.h:24]: I will address this in another commit. It is in internal review currently.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m [hunk 17]
```diff
@@ -0,0 +1,248 @@
...
+- (nullable NSArray<GADNativeAdImage *> *)images {
+  return nil;
+}
+
+- (nullable NSString *)body {
+    return _nativeAd.bodyText;
+}
+
+- (nullable GADNativeAdImage *)icon {
+    return [[GADNativeAdImage alloc] initWithImage:_nativeAd.iconImage];
```
[the remark this answers is not in the document]
Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:114]: > Is it guarantee that an image will be returned everytime?
I checked the code and if the icon URL is invalid, it may be nil. In the coming commit that addresses your feedback, I added an if-check.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m [hunk 18]
```diff
@@ -0,0 +1,257 @@
...
+    return;
+  }
+    
+  [self loadAd];
+}
+
+- (void)loadAd {
+  _nativeAd = [[VungleNativeAd alloc] initWithPlacementID:self.desiredPlacement];
+  _nativeAd.delegate = self;
+  VungleAdNetworkExtras *networkExtras = [_adConfiguration extras];
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:92]: Nit, can do `VungleAdNetworkExtras *networkExtras = _adConfiguration.extras;`

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m [hunk 19]
```diff
@@ -0,0 +1,257 @@
...
+  if (networkExtras.nativeAdOptionPosition == 1) {
+    _nativeAd.adOptionsPosition = NativeAdOptionsPositionTopLeft;
+  } else if (networkExtras.nativeAdOptionPosition == 2) {
+    _nativeAd.adOptionsPosition = NativeAdOptionsPositionTopRight;
+  } else if (networkExtras.nativeAdOptionPosition == 3) {
+    _nativeAd.adOptionsPosition = NativeAdOptionsPositionBottomLeft;
+  } else if (networkExtras.nativeAdOptionPosition == 4) {
+    _nativeAd.adOptionsPosition = NativeAdOptionsPositionBottomRight;
+  }
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:101]: Nit, perhaps we can have a `switch` statement here.

In addition, what should the value of `_nativeAd.adOptionsPosition` be if the publishers don't set this extra?

Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:101]: The position will be the default value if not set, which is the top right.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m (hunk 7 above)
Comment (joshh-devrel) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:147]: That's right, missed that one.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m (hunk 9 above)
Comment (joshh-devrel) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:151]: Quick question: Is [privacy icon in Vungle](https://support.vungle.com/hc/en-us/articles/4416477036955#display-a-native-ad-0-5) referring to the [AdChoices icon](https://support.google.com/admanager/answer/2695279)?

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m (hunk 19 above)
Comment (joshh-devrel) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:101]: Thanks Stanley.

Looking at [Vungle documentation](https://support.vungle.com/hc/en-us/articles/4416477036955#configure-the-native-ad-for-display-using-storyboard-0-6), it says:
```
Before calling the method to display the Native Ad, you must add a line of code to specify the 
position of the privacy icon. This position should be one of the available positions in the 
NativeAdOptionsPosition enum found in VungleNativeAd.h.
```

Is that not the case here?

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m [hunk 20]
```diff
@@ -0,0 +1,257 @@
...
+  /// The completion handler to call when the ad loading succeeds or fails.
+  GADMediationNativeLoadCompletionHandler _adLoadCompletionHandler;
+
+  /// The Vungle native ad.
+  VungleNativeAd *_nativeAd;
+
+  /// The ad event delegate to forward ad rendering events to the Google Mobile Ads SDK.
+  id<GADMediationNativeAdEventDelegate> _delegate;
+    
+  /// The Vungle container for the main ad
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:38]: Nit, missing period at the end.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m (hunk 19 above)
Comment (joshh-devrel) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:101]: Gentle bump @stanley-vungle @ManojBudumuru-vungle.

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m [hunk 21]
```diff
@@ -0,0 +1,257 @@
...
+- (void)didCloseAd {
+    // No-op, Vungle native ads utilize callbacks from VungleNativeAdDelegate.
+}
+
+- (void)willLeaveApplication {
+    // No-op, Vungle native ads utilize callbacks from VungleNativeAdDelegate.
+}
+
+- (nullable NSString *)bidResponse {
+  return [_adConfiguration bidResponse];
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:254]: Nit: Can do `_adConfiguration.bidResponse;`

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m (hunk 19 above)
Comment (joshh-devrel) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m:101]: Another bump.

Comment (ManojBudumuru-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m]: @joshh-devrel Updated. please check. thanks

**Review by ManojBudumuru-vungle — COMMENTED** [review]
@joshh-devrel were u able to test NAtive ads e2e after the testmode is ON. Is there anything else blocking this PR to be merged. Thanks

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.h [hunk 22]
```diff
@@ -39,5 +39,6 @@ extern const CGSize kVNGBannerShortSize;
 - (void)completeBannerAdViewForPlacementID:(nonnull id<GADMAdapterVungleDelegate>)delegate;
 - (BOOL)isSDKInitialized;
 - (nullable NSString *)getSuperToken;
+- (void)setCOPPAStatus:(nonnull NSNumber *)coppa;
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.h:42]: Shouldn't this take in boolean instead of number?

See: https://developers.google.com/admob/ios/api/reference/Classes/GADRequestConfiguration?hl=en#-tagforchilddirectedtreatment:

File: adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m [hunk 23]
```diff
@@ -419,6 +421,13 @@ - (NSString *)getSuperToken {
   return [VungleSDK.sharedSDK currentSuperToken];
 }
 
+- (void)setCOPPAStatus:(NSNumber *)coppa {
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m:424]: Ditto, shouldn't this take in boolean instead of number?
https://developers.google.com/admob/ios/api/reference/Classes/GADRequestConfiguration?hl=en#-tagforchilddirectedtreatment:

File: adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleRewardedAd.m [hunk 24]
```diff
@@ -167,20 +167,18 @@ - (void)adNotAvailable:(nonnull NSError *)error {
...
   _adLoadCompletionHandler(nil, error);
-  [GADMAdapterVungleBiddingRouter.sharedInstance removeDelegate:self];
 }
 
 - (void)trackClick {
   [_delegate reportClick];
 }
 
 - (void)rewardUser {
-  id<GADMediationRewardedAdEventDelegate> strongDelegate = _delegate;
```
Comment (joshh-devrel) [inline adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleRewardedAd.m:178]: @ManojBudumuru-vungle @stanley-vungle Can you remind me again why is this removed? If I remember correctly, you have verified that `_delegate` is already strong?

Comment (stanley-vungle) [reply adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleRewardedAd.m:178]: Because it is the same thing so there was no need to create "strongDelegate". We can just use _delegate.
